src/components/Form.js

```diff
@@ -1,56 +1,66 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
 import { addBook } from '../redux/books/books';
-import Set from './Sticker';
 
-const FormInfo = () => {
-  const { values, onChange } = Set();
-  const dispatch = useDispatch;
+const Form = () => {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const dispatch = useDispatch();
 
-  const { title, author } = values;
-  const submitHandler = () => {
-    const addNewBook = {
+  const onChange = (e) => {
+    const {
+      target: { name, value },
+    } = e;
+    if (name === 'title') setTitle(value);
+    else if (name === 'author') setAuthor(value);
+  };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const newBook = {
       id: uuidv4,
       title,
       author,
     };
-    dispatch(addBook(addNewBook));
+    dispatch(addBook(newBook));
+    setAuthor('');
+    setTitle('');
+  };
```

Comment:
- [ ] [REQUIRED] Good job so far :clap: But your new functionality responsible for adding a new book is not working as expected. Your new book is not dispatching. Please fix it in order to make the app works as expected :smile:

Reply:
Hello Codecaiine. I am really stuck on this problem. I don't know the exact wrong thing I am doing. Do you kindly point me in the right direction?